balena-docs/custom-build.md

```diff
@@ -1,3 +1,23 @@
 # Building Your Own Image
 
-To build your own image, refer to [Customer Board Support](https://www.balena.io/docs/reference/OS/customer-board-support/) section of docs. 
\ No newline at end of file
+To build your own custom balenaOS image, refer to the README for respective device type repository hosted in the [balenaOS GitHub org](https://github.com/balena-os/) for more information.
+
+Identifying the right repository is important. All device type repositories will start with `balena-*`. As we build device types of similar specification or board manufacturers together inside one repository. For example, to find the right repository for the device type Raspberry Pi 3. 
+
+1. One could search for the term `raspberry` on the [GitHub Org search page](https://github.com/balena-os?q=raspberry&type=all&language=&sort=)
+2. From the results, we will find the repository called [balena-raspberrypi](https://github.com/balena-os/balena-raspberrypi)
+3. In the repository, we will find a coffeescript file by the name [raspberrypi3.coffee](https://github.com/balena-os/balena-raspberrypi/blob/master/raspberrypi3.coffee). 
+
+This proves that the [balena-raspberrypi](https://github.com/balena-os/balena-raspberrypi) is in fact that the place to find build instructions for Raspberry Pi 3 device. Most coffeescript file are named based on [Device type slugs](https://docs.balena.io/reference/hardware/devices/) which are helpful to know beforehand.
+
+Similarly here are some examples for other popular devices. 
+
+| Device Type     | GitHub Repository                                                                              |
+| --------------- | ---------------------------------------------------------------------------------------------- |
+| `intel-nuc`     | [https://github.com/balena-os/balena-intel](https://github.com/balena-os/balena-intel)         |
+| `jetson-nano`   | [https://github.com/balena-os/balena-jetson](https://github.com/balena-os/balena-jetson)       |
+| `imx7-var-som`  | [https://github.com/balena-os/balena-variscite](https://github.com/balena-os/balena-variscite) |
+| `nanopi-r2s`    | [https://github.com/balena-os/balena-nanopi-r2](https://github.com/balena-os/balena-nanopi-r2) |
+| `generic-amd64` | [https://github.com/balena-os/balena-generic](https://github.com/balena-os/balena-generic)     |
+| `odyssey-x86`   | [https://github.com/balena-os/balena-seeed-x86](https://github.com/balena-os/balena-seeed-x86) |
+
```

Comment:
@vipulgupta2048 this feels complicated, let's simplify it. What I would do is instruct to use the balenaCLI to find the machine name from the slug, which is what users would typically know from balenaCloud:
```
balena devices supported | grep $SLUG
```
And then just [search](https://github.com/search?q=org%3Abalena-os%20raspberrypi3.coffee&type=code) the balenaos org for <machine name.coffee>.

Then refer to the [Build information](https://github.com/balena-os/balena-raspberrypi/blob/master/README.md#build-information) section of the README:

That always works - also we may not need the table - I know I proposed this but it just feels too much.

Reply:
@floion Added back that line.
@alexgg Not sure if asking someone to use (and install) another tool to just find device slugs is better than a webpage. It is complicated, I found the same when I was writing it in the first place. 

I would agree looking for repos might be a touch extra. Will revert to searching the org for files.

Reply:
> if asking someone to use (and install) another tool to just find device slugs is better than a webpage

imo it's better because it always works - not all device types have matching device types and slugs. It's the tool we have, and in the context of doing custom builds I would not think using another balena tool is too much to ask.

Reply:
I feel we can leave the table in there for some reference

Reply:
Is balena-cli required to build balenaOS images? @alexgg I think the web pages feels more complete don't you think? At this point, I am not sure why we are trying add one more tool when the user is just looking for docs on how to do something?

Reply:
For a minute there, I forgot the entire thing about machine name. Changing it your CLI way.

Reply:
@alexgg Using the CLI command doesn't yield machine names, only slugs. 

```
➜  meta-balena git:(vipul/custom-build) balena devices supported | grep "intel-nuc"
intel-nuc                              nuc                    amd64   Intel NUC
```

Reply:
you're right @vipulgupta2048 I was hoping the `ALIASES` column would list machine names - not the case.

Searching for `coffee` language in the `balena_os` org with the slug name seems to work and provide single repository matches though:
https://github.com/search?q=org%3Abalena-os+language%3Acoffee+%22Intel+nuc%22&type=code

Reply:
Done, check the changes and let me know what you think @alexgg 
A customer asked a question about this on support so remembered to come back to it.